crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -253,6 +253,74 @@ impl<'w> EntityMut<'w> {
         self
     }
 
+    /// Inserts a component with the given `value`. Will replace the value if it already existed.
+    ///
+    /// **You should prefer to use the typed API [`EntityMut::insert`] where possible and only
+    /// use this in cases where the actual types are not known at compile time.**
```

Comment:
I think this warning is too generous, we want to be "stricter" than just "are not known at compile time." rust already has `dyn Trait` (for example bevy supports `Box<dyn ApplicableBundle>`) for cases like this. I'd prefer:
```suggestion
    /// use this in cases where there isn't a Rust type corresponding to the `ComponentId`.**
```